src/main/java/io/tus/java/client/TusUploader.java

```diff
@@ -280,6 +280,30 @@ public void finish() throws ProtocolException, IOException {
         // that we will not need to read from it again in the future.
         input.close();
     }
+    
+    /**
+     * Finish the request by closing the HTTP connection. You can choose whether to close the Input Stream or not.
+     * You can call this method even before the entire file has been uploaded.
+     * It pauses the upload properly to be resumed in future.
+     * Be aware it doesn't release local resources as it does not close the File's
+     * Input Stream if {@code closeStream == false}
+     * To be safe use {@link TusUploader#finish()}.
+     * @param closeInputStream Determines whether the InputStream is closed with the HTTP connection. Not closing the
+     *                         Input Stream may be useful for future upload a future continuation of the upload.
+     * @throws ProtocolException Thrown if the server sends an unexpected status code
+     * @throws IOException  Thrown if an exception occurs while cleaning up.
+     */
+    public void finish(boolean closeInputStream) throws ProtocolException, IOException {
+        if (closeInputStream){
+            finish();
+        } else {
+            finishConnection();
+            if (upload.getSize() == offset) {
+                client.uploadFinished(upload);
+            }
```

Comment:
We are duplicating some logic here now. The lines of 
```java
            finishConnection();
            if (upload.getSize() == offset) {
                client.uploadFinished(upload);
            }
```

are now present in two method. In order to DRY up the code (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself in case you haven't seen this acronym yet :), I could image having the `finish(boolean closeInputStream)` method similar to its current state where it contain the logic of (possibly) closing the input stream and the connection. Then the older `finish()` method can just be a call to the new method using `finish(false)`. What do you think?